prow/config/inrepoconfig.go

```diff
@@ -129,8 +130,14 @@ func prowYAMLGetter(
 
 func ensureHeadCommits(repo git.RepoClient, headSHAs ...string) error {
 	for _, sha := range headSHAs {
-		if err := repo.Fetch(sha); err != nil {
-			return fmt.Errorf("failed to fetch headSHA: %s: %v", sha, err)
+		exists, err := repo.CommitExists(sha)
```

Comment:
this is a best effort imo to avoid fetching when it returned true, and fetch if not, do we really need to check error in this case?

Reply:
Done